cmd/kube-apiserver/app/options/options_test.go

```diff
@@ -17,26 +17,167 @@ limitations under the License.
+	BindAddress net.IP
+	// --client-ca-file
+	ClientCAFile string
+	// --cloud-config
+	CloudConfig string
+	// --cloud-provider
+	CloudProvider string
+	// --cors-allowed-origins
+	CorsAllowedOrigins []string
+	// --enable-aggregator-routing
+	EnableAggregatorRouting bool
+	// --enable-logs-handler
+	EnableLogsHandler bool
+	// --enable-swagger-ui
+	EnableSwaggerUI bool
+}
+
 func TestAddFlagsFlag(t *testing.T) {
-	// TODO: This only tests the enable-swagger-ui flag for now.
-	// Expand the test to include other flags as well.
+	// TODO: Expand the test to include other flags as well.
 	f := pflag.NewFlagSet("addflagstest", pflag.ContinueOnError)
 	s := NewServerRunOptions()
 	s.AddFlags(f)
-	if s.Features.EnableSwaggerUI {
-		t.Errorf("Expected s.EnableSwaggerUI to be false by default")
-	}
 
 	args := []string{
+		"--advertise-address=192.168.10.10",
+		"--allow-privileged=false",
+		"--anonymous-auth=false",
+		"--apiserver-count=5",
+		"--audit-log-maxage=11",
+		"--audit-log-maxbackup=12",
+		"--audit-log-maxsize=13",
+		"--audit-log-path=/var/log",
+		"--audit-policy-file=/policy",
+		"--audit-webhook-config-file=/webhook-config",
+		"--audit-webhook-mode=blocking",
+		"--authentication-token-webhook-cache-ttl=3m",
+		"--authentication-token-webhook-config-file=/token-webhook-config",
+		"--authorization-mode=AlwaysDeny",
+		"--authorization-policy-file=/policy",
+		"--authorization-webhook-cache-authorized-ttl=3m",
+		"--authorization-webhook-cache-unauthorized-ttl=1m",
+		"--authorization-webhook-config-file=/webhook-config",
+		"--bind-address=192.168.10.20",
+		"--client-ca-file=/client-ca",
+		"--cloud-config=/cloud-config",
+		"--cloud-provider=daocloud",
+		"--cors-allowed-origins=10.10.10.100,10.10.10.200",
+		"--enable-aggregator-routing=true",
+		"--enable-logs-handler=false",
 		"--enable-swagger-ui=true",
 	}
 	f.Parse(args)
-	if !s.Features.EnableSwaggerUI {
-		t.Errorf("Expected s.EnableSwaggerUI to be true")
+
+	// Kubernetes apiserver run options
+	parsed := Options{
+		s.GenericServerRunOptions.AdvertiseAddress,
+		s.AllowPrivileged,
+		s.Authentication.Anonymous.Allow,
+		s.MasterCount,
+		s.Audit.LogOptions.MaxAge,
+		s.Audit.LogOptions.MaxBackups,
+		s.Audit.LogOptions.MaxSize,
+		s.Audit.LogOptions.Path,
+		s.Audit.PolicyFile,
+		s.Audit.WebhookOptions.ConfigFile,
+		s.Audit.WebhookOptions.Mode,
+		s.Authentication.WebHook.CacheTTL,
+		s.Authentication.WebHook.ConfigFile,
+		s.Authorization.Mode,
+		s.Authorization.PolicyFile,
+		s.Authorization.WebhookCacheAuthorizedTTL,
+		s.Authorization.WebhookCacheUnauthorizedTTL,
+		s.Authorization.WebhookConfigFile,
+		s.SecureServing.BindAddress,
+		s.Authentication.ClientCert.ClientCA,
+		s.CloudProvider.CloudConfigFile,
+		s.CloudProvider.CloudProvider,
+		s.GenericServerRunOptions.CorsAllowedOriginList,
+		s.EnableAggregatorRouting,
+		s.EnableLogsHandler,
+		s.Features.EnableSwaggerUI,
+	}
+
+	// Expected apiserver run options set by args
+	expected := Options{
```

Comment:
What about converting this `Options` instance in the format of `s`? Then we don't need our local `Options` type. Makes much easier to maintain.